### PR TITLE
Do not make up decimal places

### LINKED ISSUE #8155: `$significant_decimals` generates noise
1. disable `intl`
2. view the query log
3. chaos ensures now that it uses `unlimied-precision`

Screenshot 2024-03-28 at 16-42-06 My Community

5. the second image shows the result when `intl` is enabled again

Screenshot 2024-03-28 at 16-40-03 My Community

### PR DESCRIPTION
**Pull request by live627**
Lots of noise is generated when using the stem `unlimited-precision`

### CODE REVIEW

**Comment by Sesquipedalian** [comment]
Might as well change the code on this line, too.
https://github.com/SimpleMachines/SMF/blob/a182d918c1db9c6c87f84adfeaafdefe14453f40/Sources/Localization/MessageFormatter.php#L868

**Comment by live627** [comment]
added, unsure if it's right or how to test if there is no off by one error.

File: Sources/Localization/MessageFormatter.php [hunk 1]
```diff
@@ -865,7 +865,7 @@ protected static function applyNumberSkeleton(int|float|string $number, string $
 
 		// Ensure $number is a string.
 		if (is_float($number)) {
-			$precision = max(0, strlen(rtrim(sprintf('%.53F', abs($number + 0)), '0')) - strlen(strval(intval($number + 0))) - 1);
+			$precision = (int) strpos(strrev(strval($number)), '.') - strlen(strval(intval($number + 0))) - 1;
```
Comment (Sesquipedalian) [inline Sources/Localization/MessageFormatter.php]:
```suggestion
			$precision = (int) strpos(strrev(strval($number)), '.');
```

^ This is what you replaced `max(0, strlen(rtrim(sprintf('%.53F', abs($number + 0)), '0')) - strlen(strval(intval($number + 0))) - 1);` with in the other two cases, so I would think it is what we need in this case as well. Or am I missing something here?

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

File: Sources/Localization/MessageFormatter.php (hunk 1 above)
Comment (live627) [reply Sources/Localization/MessageFormatter.php]: Probably. I was unsure on this, so wanted your input.

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**